### PR TITLE
Source Control Management Docs

### PR DESCRIPTION
**Pull request by ryanmillerdev**
A couple of final touches to add (images), but wanted to get this in front of folks with the intention of having these ready to go immediately after the new year.

This PR introduces a new page dedicated to _source control managment_, which replaces [these docs](https://algorithmia.com/developers/algorithm-development/git), given that our "internal" repository hosting solution is no longer the sole solution we offer to users.

Will deploy to test once the final image makes it in.

Docs for admins are being worked on separately in [this Google Doc](https://docs.google.com/document/d/1uqVP3muwhYpYR9YiEAz3vwMEB_D8dbiMgEhWmDqqXjE/edit#heading=h.ww85iwv7bf85), with their home ultimately being Skilljar (training.algorithmia.com).

### CODE REVIEW

File: _pages/algorithm-development/source-control-management.md
Comment (jamesatha) [inline _pages/algorithm-development/source-control-management.md]: Can we change the last `username` to `name`? 
The fact that organizations have usernames is an implementation detail we need to tell people and it's confusing

File: _pages/algorithm-development/source-control-management.md
Comment (jamesatha) [inline _pages/algorithm-development/source-control-management.md:23]: Should this be in a `{% if site.enterprise %}` block?

File: _pages/algorithm-development/source-control-management.md
Comment (jamesatha) [inline _pages/algorithm-development/source-control-management.md]: I feel like we shouldn't have to explain this

File: _pages/algorithm-development/source-control-management.md
Comment (jamesatha) [inline _pages/algorithm-development/source-control-management.md]: Kind of related to my previous comment, I would say that we use this to get source code and commit messages with the deploy key here. And maybe state that these should remain after people leave/removed from an organization. 
Do we also want to mention some of the stuff you noticed: when you remove an oauth token from the UI, it removes the deploy key which can break functionality

File: _pages/algorithm-development/source-control-management.md
Comment (jamesatha) [inline _pages/algorithm-development/source-control-management.md]: do we want to specify that we look at the default branch which doesn't need to be master? 

And if we get the hmac secret stuff done, is this where we would update the docs? (Hoping to have that out for review soon)

**Review by jamesatha — APPROVED** [review]
Sweet! I've got some suggestions but none that would block this from going in

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md]: Could I pick your brain a bit more here? Given we mediate GitHub data access via OAuth tokens in all other cases, I do feel it's worth covering our bases here and mentioning the outlier cases?

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md:23]: Good call, I'll add.

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md]: Ah, I get what you mean. I like that. And I'll make sure to add that item to the FAQ!

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md]: (Followed up below, I like the idea of repositioning this to the deploy key snippet)

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md]: Will add!

File: _pages/algorithm-development/source-control-management.md
Comment (open-abbott) [inline _pages/algorithm-development/source-control-management.md:150]: Were you surrounding all off the "Enterprise Users" notes with a conditional block?

File: _pages/algorithm-development/source-control-management.md
Comment (stevielkim) [inline _pages/algorithm-development/source-control-management.md]: This doc needs a table of contents as it's fairly long with defined sections like: "Hosting Code On Algorithmia" and the sub sections coming under that and the next section: "Hosting Algorithm Source Code on GitHub".

Also before you get into the hosting on algorithmia then hosting on github just a quick preface about why you would want to do one versus the other - laying out use cases is really helpful for users. It also tells the reader what the doc covers.

File: _pages/algorithm-development/source-control-management.md
Comment (stevielkim) [inline _pages/algorithm-development/source-control-management.md]: should be "git add" not git -A which isn't a git command but a flag so it could be "git add -A" or "-all" flag or of course "git add ." or "git add filename" I think this was copied from old git pages that Ahmad created forever ago and wasn't caught before.

**Review by stevielkim — CHANGES_REQUESTED** [review]
Any other issues I already communicated in this doc https://docs.google.com/document/d/1R0zuT-zuW3VD69BSN3oxFdJFR9Mf4tQXl7WWErrLscQ/edit?usp=sharing

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md]: I can add this!

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md]: Good catch!

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md:150]: Good catch! Added.

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [reply _pages/algorithm-development/source-control-management.md]: @StephanieKim Added!

**Comment by ryanmillerdev** [comment]
@StephanieKim I believe I've addressed your concerns–if you'd like to give this one more pass before we merge later today, would be appreciative!

File: _pages/algorithm-development/source-control-management.md
Comment (ryanmillerdev) [inline _pages/algorithm-development/source-control-management.md]: ✍️ Typo noted! (colleages -> colleagues)

Comment (stevielkim) [reply _pages/algorithm-development/source-control-management.md]: Perfect! Thanks!

**Pull request merged**
